### ISSUE TITLE
Adding checkstyle write-mode?

### ISSUE THREAD

**Issue by markstory** [rust-lang-nursery/rustfmt#731]
I couldn't find any previous issues about this, but was wondering if there was any interest in adding a checkstyle write mode?

Checkstyle is an XML output format that is supported by a number of continuous integration tools like Jenkins, and would allow rustfmt to easily integrate with plugins like [Violations](https://wiki.jenkins-ci.org/display/JENKINS/Violations) or [Checkstyle](https://wiki.jenkins-ci.org/display/JENKINS/Checkstyle+Plugin). It would also provide a machine readable output for `rustfmt`.

If there is interest in this, I'm happy to try and implement this write mode.

**Comment by nrc** [comment]
Sure, that sounds like it would be great to have!

**Comment by markstory** [comment]
Cool, I'll try to cobble something together.

**Comment by sinhpham** [comment]
This is related to https://github.com/rust-lang-nursery/rustfmt/issues/656

Regarding the format, unified diff format is a popular one: https://www.artima.com/weblogs/viewpost.jsp?thread=164293

**Comment by markstory** [comment]
I've made some progress on this. I just have to address XML entity encoding and then I will be able to put a pull request up for review. I'm also happy to poke at outputting a unified diff format after getting a yes/no on the checkstyle output.

**Comment by markstory** [comment]
I think this can be closed now.

**Comment by messense** [comment]
This is great.

But when Rustfmt failed, it writes the logs to stdout which makes it hard to parse checkstyle xml ouput because checkstyle output is on stdout too.

![image](https://cloud.githubusercontent.com/assets/1556054/13385294/fb56adce-ded9-11e5-92ee-7c122552eb27.png)

**Comment by messense** [comment]
BTW, I think when write mode is checkstyle, rustfmt should act like a code linter and it shouldn't fail when line exceeded maximum length occurred.

**Comment by markstory** [comment]
@messense The fail messages must be a more recent change. I don't remember seeing those when I worked on the checkstyle output.
